### PR TITLE
added /user/getAllOf end point and update API doc

### PR DESCRIPTION
**Pull request by sou-c**

### CODE REVIEW

File: src/eaeInterface.js [hunk 1]
```diff
@@ -180,6 +180,7 @@ EaeInterface.prototype._setupInterfaceControllers = function() {
     // Manage the users who have access to the platform - Admin only
     _this.app.post('/user/', _this.usersController.getUser)
         .post('/user/create', _this.usersController.createUser)
+        .post('/user/getAllOf', _this.usersController.getAllUsers)
```
Comment (aoehmichen) [inline src/eaeInterface.js]: getAllOf sounds very weird and doesn't convey what the route actually does.

Can you please rename it to getAllUsers ?

File: test/users.test.js [hunk 2]
```diff
@@ -182,6 +208,84 @@ test('Create a new user', function(done) {
...
+    request(
+        {
+            method: 'POST',
+            baseUrl: 'http://127.0.0.1:' + config.port,
+            uri: '/user/getAllOf',
+            json: true,
+            body: {
+                eaeUsername: adminUsername,
+                eaeUserToken: adminPassword,
+                userType: 'admin'
```
Comment (aoehmichen) [inline test/users.test.js]: try to be consistent in the naming convention. you use "STANDARD" above then "all" and "admin" in lower cases

Comment (sou-c) [reply test/users.test.js]: testing the conversion from lower case to upper in the function

Comment (aoehmichen) [reply test/users.test.js]: then write it in the description of the test because that's no obvious :)

File: src/eaeInterface.js (hunk 1 above)
Comment (sou-c) [reply src/eaeInterface.js]: done. thought /user/getAll would be better since it s consistent with /job/getAll

File: src/eaeInterface.js [hunk 3]
```diff
@@ -180,7 +180,7 @@ EaeInterface.prototype._setupInterfaceControllers = function() {
     // Manage the users who have access to the platform - Admin only
     _this.app.post('/user/', _this.usersController.getUser)
         .post('/user/create', _this.usersController.createUser)
-        .post('/user/getAllOf', _this.usersController.getAllUsers)
+        .post('/user/getAll', _this.usersController.getAllUsers)
```
Comment (aoehmichen) [inline src/eaeInterface.js]: the naming getAll is not much better

**Comment by aoehmichen** [comment]
OK the new name is better. Indeed same convention

**Comment by sou-c** [comment]
:)

**Pull request merged**